### ISSUE TITLE
Wrong output for GAFeatureSelectionCV only when using max_features for RandomForestClassifier

### ISSUE THREAD

**Issue by cewinharhar** [rodrigo-arenas/Sklearn-genetic-opt#88]
**System information**
OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10.0.19044
Sklearn-genetic-opt  version: 0.8.0
Scikit-learn version: 1.0.1
Python version: 3.8.11

**Describe the bug**

Firstly, great work!! 

When using :
```python
clf_RF    = RandomForestClassifier(random_state=0)

evolved_estimator = GAFeatureSelectionCV(
    estimator   = clf_RF,
    cv          = 5,
    population_size=20, 
    generations =40,
    crossover_probability=0.8,
    mutation_probability = 0.075,
    n_jobs      = -1,
    scoring     = "accuracy",
    max_features = 300
    )

# Train and select the features
evolved_estimator.fit(X, y)
```

the Output looks like this;

```
gen	nevals	fitness	fitness_std	fitness_max	fitness_min
0  	20    	-10000 	0          	-10000     	-10000     
1  	32    	-10000 	0          	-10000     	-10000     
2  	33    	-10000 	0          	-10000     	-10000     
3  	37    	-10000 	0          	-10000     	-10000     
4  	36    	-10000 	0          	-10000     	-10000     
5  	37    	-10000 	0          	-10000     	-10000     
6  	36    	-10000 	0          	-10000     	-10000     
7  	36    	-10000 	0          	-10000     	-10000     
8  	36    	-10000 	0          	-10000     	-10000     
9  	33    	-10000 	0          	-10000     	-10000     
10 	33    	-10000 	0          	-10000     	-10000     
11 	34    	-10000 	0          	-10000     	-10000     
12 	34    	-10000 	0          	-10000     	-10000     
13 	36    	-10000 	0          	-10000     	-10000     
14 	33    	-10000 	0          	-10000     	-10000     
15 	34    	-10000 	0          	-10000     	-10000     
16 	33    	-10000 	0          	-10000     	-10000     
17 	37    	-10000 	0          	-10000     	-10000     
18 	35    	-10000 	0          	-10000     	-10000     
19 	37    	-10000 	0          	-10000     	-10000     
20 	34    	-10000 	0          	-10000     	-10000     
21 	35    	-10000 	0          	-10000     	-10000     
22 	35    	-10000 	0          	-10000     	-10000 
```

This doesn't happen when removing the max_feature() parameter.

**To Reproduce**
2. Add this code '....'
3. Run with this command '....'

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by rodrigo-arenas** [comment]
Hi, how many features have your X? Is this public data you can share?

The model uses -10000 as a penalization value when it assigns 0 features or a value greater than the max_features, this is the part of the code that handles this value:

```python
if self.max_features and (n_selected_features > self.max_features or n_selected_features == 0):
            score = -self.criteria_sign * 10000
```

**Comment by cewinharhar** [comment]
> Hi, how many features have your X? Is this public data you can share?
> 
> The model uses -10000 as a penalization value when it assigns 0 features or a value greater than the max_features, this is the part of the code that handles this value:
> 
> ```python
> if self.max_features and (n_selected_features > self.max_features or n_selected_features == 0):
>             score = -self.criteria_sign * 10000
> ```

Hi Rodrigo! 
Thanks for the quick reply.
X has the shape 181, 2730 and therefore 2730 features. The data will be published in 2 weeks so no its not yet public. But for a snippet:

``` 
X[:10, :10]

array([[13.86750423, 15.44779027, 12.36815207, 11.41035081, 12.65277529,
        12.27863945, 13.17554833, 15.30203768, 16.21976425, 15.61555802],
       [15.23465169, 15.65745726, 13.14293189, 12.96310263, 12.44629581,
        11.7719609 , 12.69401105, 16.62334569, 19.97297462, 13.90634831],
       [12.08458637, 15.49691959, 12.9215506 , 12.64248375, 11.50311202,
        11.68014723, 12.87054816, 13.93805251, 15.29687967, 16.11543488],
       [12.10872624, 12.1352828 , 12.31044554, 12.45658729, 11.93978033,
        11.88551932, 12.08187331, 11.51125946, 13.54278003, 14.67511014],
       [15.10646789, 15.42363273, 13.256691  , 12.30760593, 11.79079909,
        12.34581927, 12.24209265, 14.25753589, 17.22163874, 15.83668828],
       [17.35827479, 18.26202113, 15.42031904, 12.92403792, 14.8273144 ,
        16.12210379, 16.794989  , 17.32657363, 19.99976626, 18.74650776],
       [14.90420342, 15.77959754, 14.43681598, 11.80109051, 12.94046227,
        12.47073086, 12.92760374, 13.80184589, 15.5056026 , 16.39919603],
       [14.7953891 , 15.13048745, 11.41319043, 11.77437477, 12.83703548,
        12.95054504, 12.5048368 , 13.80259781, 17.20993268, 15.14166438],
       [15.19413042, 16.69400743, 11.98449964, 11.92335587, 12.44016291,
        13.92653616, 13.24677727, 15.24775487, 17.5494733 , 15.51427714],
       [13.16452766, 17.04205058, 13.34011795, 11.51953644, 12.54358967,
        12.32742861, 12.45736823, 14.18855927, 17.27017326, 15.39236866]])

y[:10]

array([4, 1, 3, 4, 0, 2, 4, 3, 1, 2])

The GAFeatureSelectionCV is working with max_feature = 1500 but not with 500.So I assume that max_features must be greater than n_selected_features to handle this.
```

**Comment by rodrigo-arenas** [comment]
That is right, max_features must be greater than n_selected_features; you can also try to increase the `mutation_probability` parameter to around 0.2.
I'll try to simulate this volume of data and see if I can reproduce this, I'll also take a look when the data is publicly available.

**Comment by rodrigo-arenas** [comment]
Hi @cewinharhar 
I took a closer look into this, and I realized what could be happening:

At generation 0, we have N solutions (population size), each solution is a random sample of the features, this random sample may or may not have less than max_features since it follows a uniform distribution.
I simulated around 3k variables using max_features=300, this lead to some runs where there is no random solution with less than 300 features, so through cross-over and mutation its' hard to get a new solution in this range.

I'm already working on a fix and It will be released in the next version, hopefully, this week.
To summarise the change I'll introduce: if max_features is not None, now at generation 0, each feature has a probability of being sampled of max_features/total_features. With this, it will not get stuck on non-feasible solutions.

![image](https://user-images.githubusercontent.com/31422766/149346299-09f67a5c-6a1d-4b4d-8b59-619794889b24.png)

**Comment by rodrigo-arenas** [comment]
The solution previously explained now is available in version 0.8.1 via PR #92
